js/ui/components/clubhouse.js

```diff
@@ -148,6 +153,111 @@ var ClubhouseWindowTracker = new Lang.Class({
     },
 });
 
+var getClubhouseAnimator = (function () {
+    let clubhouseAnimator;
+    return function() {
+        if (!clubhouseAnimator)
+            clubhouseAnimator = new ClubhouseAnimator();
+        return clubhouseAnimator;
+    };
+}());
+
+var ClubhouseAnimator = new Lang.Class({
+    Name: 'ClubhouseAnimator',
+    Extends: GObject.Object,
+
+    Signals: {
+        'animation-ready': {},
+    },
+
+    _init: function() {
+        this.parent();
+        this._proxy = null;
+        this._animations = {};
+        this._clubhousePaths = this._getClubhousePaths();
+    },
+
+    setClubhouseProxy: function(proxy) {
+        this._proxy = proxy;
+    },
+
+    _getClubhousePaths: function() {
+        let paths = [];
+        let installations = Flatpak.get_system_installations(null);
+        installations.unshift(Flatpak.Installation.new_user(null));
+
+        for (let installation of installations) {
+            let app = null;
+            try {
+                app = installation.get_current_installed_app(CLUBHOUSE_ID, null);
+            } catch (err if err.matches(Flatpak.Error, Flatpak.Error.NOT_INSTALLED)) {
+                continue;
+            }
+
+            if (app) {
+                let deploy_dir = app.get_deploy_dir();
+                paths.push(this._getActivateDir(deploy_dir));
+            }
+        }
+
+        return paths;
+    },
+
+    _getActivateDir: function(deployDir) {
+        // Replace the hash part of the deploy directory by "active", so the directory
+        // is always the most recent one (i.e. allows us to update the Clubhouse and
+        // still have the right dir).
+        let dir = deployDir.substr(-1) == '/' ? deployDir.slice(0, -1) : deployDir;
+        let splitDir = dir.split('/')
+
+        splitDir[splitDir.length - 1] = 'active';
+
+        return splitDir.join('/');
+    },
+
+    _getClubhouseUri: function(uri) {
+        // Discard the /app/ prefix
+        let uriSuffix = uri.replace(/^\/app\//g, '');
+
+        for (let path of this._clubhousePaths) {
+            let completeUri = path + '/files/' + uriSuffix;
```

Comment:
This is a path, not an uri... Also consider using e.g. `GLib.build_filenamev()` to build the path.